python/dgl/contrib/sampling/sampler.py

```diff
@@ -300,26 +320,48 @@ def __init__(
 
         assert g.is_readonly, "NeighborSampler doesn't support mutable graphs. " + \
                 "Please turn it into an immutable graph with DGLGraph.readonly"
-        assert node_prob is None, 'non-uniform node probability not supported'
         assert isinstance(expand_factor, Integral), 'non-int expand_factor not supported'
 
         self._expand_factor = int(expand_factor)
         self._num_hops = int(num_hops)
         self._add_self_loop = add_self_loop
         self._num_workers = int(num_workers)
         self._neighbor_type = neighbor_type
+        self._transition_prob = transition_prob
 
     def fetch(self, current_nodeflow_index):
-        handles = unwrap_to_ptr_list(_CAPI_UniformSampling(
-            self.g.c_handle,
-            self.seed_nodes.todgltensor(),
-            current_nodeflow_index, # start batch id
-            self.batch_size,        # batch size
-            self._num_workers,      # num batches
-            self._expand_factor,
-            self._num_hops,
-            self._neighbor_type,
-            self._add_self_loop))
+        if self._transition_prob is None:
+            prob = None
+        elif isinstance(self._transition_prob, str):
+            prob = self.g.edata[self._transition_prob]
```

Comment:
Add a check to make sure the edge feature is scalar.

Reply:
It's done in C.